lib/src/Timeline/TimelineItem.tsx

```diff
@@ -0,0 +1,54 @@
+import React from "react";
+import classnames from "classnames";
+import ResizeObserver from "resize-observer-polyfill";
+
+export type VerticalTimelineItemPosition = "left" | "right";
+export type HorizontalTimelineItemPosition = "top" | "bottom";
+export type TimelineDirection = "vertical" | "horizontal";
+
+export type TimelineItemProps = JSX.IntrinsicElements["div"] & {
+    wrapperProps?: JSX.IntrinsicElements["div"];
+    title: React.ReactNode;
+    time: string;
+    desc?: React.ReactNode;
+    direction?: TimelineDirection;
+    position?: VerticalTimelineItemPosition | HorizontalTimelineItemPosition;
```

Comment:
Since the direction and position is something controlled by the parent, there is no need to expose a prop for it. The best way I found so far is to pass `data-position` from the parent. In this case, it will not be documented and the user won't know it existed since it's only meant for internal use.

You can remove the direction and position props and just pass it from the parent. The other good part about it is that you can use CSS to apply styles directly to it so you don't need to define a class for it.